### ISSUE TITLE
[Bug] - Changing Scheduler to SCHED_RR not working

### ISSUE THREAD

**Issue by goznauk** [amazonlinux/amazon-linux-2023#315]
**Describe the bug**
Changing Scheduler to SCHED_RR or SCHED_FIFO is not working. 
`chrt` `sched_setscheduler` or `CPUSchedulingPolicy=rr` in systemd service all fails because of `Operation not permitted`.
I've tried it with `sudo` or `root` or `sudo setcap cap_sys_nice+ep "$(readlink -f $(which something))"` but none of them worked.

**To Reproduce**
```
$ sudo chrt --rr 10 ls
```

This fails with error `chrt: failed to set pid 0's policy: Operation not permitted`

**Expected behavior**
I've expected run with root will success, but it also fails. Is it bug?

**Additional context**
I'm novice linux user, so please notice me if I've missed something.

**Comment by goznauk** [comment]
I've found why. It's because of `CONFIG_RT_GROUP_SCHED` is enabled, unlike other common general-purpose kernels.

**Comment by ozbenh** [comment]
This is correct, we are looking into it. There are ways to specify time allocation to the cgroups created by `systemd`, but it's cumbersome and it appears that other distributions disable `CONFIG_RT_GROUP_SCHED` instead. We're investigating what the best option is for AL2023. We'll either turn this off in a future release or document how to "work around" with `systemd`

**Comment by goznauk** [comment]
@ozbenh Thanks for reply!

[Readme](https://github.com/systemd/systemd/blob/e8dc52766e1fdb4f8c09c3ab654d1270e1090c8d/README#L138) in `systemd` repository says that

> We recommend to turn off Real-Time group scheduling in the
> kernel when using systemd. RT group scheduling effectively
> makes RT scheduling unavailable for most userspace, since it
> requires explicit assignment of RT budgets to each unit whose
> processes making use of RT. As there's no sensible way to
> assign these budgets automatically this cannot really be
> fixed, and it's best to disable group scheduling hence.
>    CONFIG_RT_GROUP_SCHED=n


and very naive quick fix for this is [disable RT throttling](https://wiki.linuxfoundation.org/realtime/documentation/technical_basics/sched_rt_throttling)

```sh
sudo sysctl -w kernel.sched_rt_runt ime_us=-1
```

Is this "safe" if I can control "not to starve"? or should I avoid this?
I'm willing to have some "safe" workarounds, or next AL2023 build without `CONFIG_RT_GROUP_SCHED` :)

**Comment by kuzaxak** [comment]
@ozbenh Hey, do you have any news on it? We faced the same issue.

**Comment by ozbenh** [comment]
It looks like our kernel team might have dropped the ball on that one. I'll poke internally.

**Comment by yifei-aws** [comment]
In addition, when you ran the `chrt`, is the user in any cgroup?
```
cat /proc/self/cgroup
0::/user.slice/user-1000.slice/session-1.scope
```

**Comment by yifei-aws** [comment]
OK, read the original report issue again. What I read is you are not complaining a process can't be set to be scheduled by real-time scheduler, like your simple reproducer, you are saying you are not able to set to a task in a systemd cgroups to the real-time scheduler.
FYI, real-time scheduler is always available not behind any configuration.

So, can you provide more about your systemd cgroups ?
